src/components/DeploymentList.vue

```diff
@@ -0,0 +1,352 @@
+            <p-loading-icon />
+          </template>
+        </PEmptyResults>
+      </template>
+    </p-table>
+
+    <p-pager v-if="pages > 1" v-model:page="page" :pages="pages" />
+  </p-content>
+</template>
+
+<script lang="ts" setup>
+  import { TableColumn, media } from '@prefecthq/prefect-design'
+  import { NumberRouteParam, useDebouncedRef, useRouteQueryParam } from '@prefecthq/vue-compositions'
+  import { secondsInWeek } from 'date-fns/constants'
+  import { snakeCase } from 'lodash'
+  import merge from 'lodash.merge'
+  import { computed, ref } from 'vue'
+  import {
+    DeploymentsDeleteButton,
+    ResultsCount,
+    SearchInput,
+    FlowPopover,
+    MiniDeploymentHistory,
+    SelectedCount,
+    DeploymentTagsInput,
+    DeploymentToggle,
+    FormattedDate,
+    DeploymentStatusBadge
+  } from '@/components'
+  import { useCan, useDeploymentsFilterFromRoute, useWorkspaceRoutes, useDeployments, useComponent } from '@/compositions'
+  import { Deployment, DeploymentSchedule } from '@/models'
+  import { DeploymentsFilter } from '@/models/Filters'
+  import { ClassValue } from '@/types'
+  import { deploymentSortOptions } from '@/types/SortOptionTypes'
+
+  const props = defineProps<{
+    filter?: DeploymentsFilter,
+    prefix?: string,
+  }>()
+
+  const { DeploymentMenu } = useComponent()
+
+  const can = useCan()
+  const routes = useWorkspaceRoutes()
+
+  const nameLike = ref<string>()
+  const nameLikeDebounced = useDebouncedRef(nameLike, 1200)
+
+  const { filter, clear, isCustomFilter } = useDeploymentsFilterFromRoute(merge({}, props.filter, {
+    deployments: {
+      flowOrDeploymentNameLike: nameLikeDebounced,
+    },
+    limit: 50,
+  }), props.prefix)
+
+  const page = useRouteQueryParam('page', NumberRouteParam, 1)
+  const { deployments, subscriptions, total, pages } = useDeployments(filter, {
+    page,
+    interval: 30000,
+  })
+
+  const columns = computed<TableColumn<Deployment>[]>(() => [
+    {
+      label: 'Deployment',
+    },
+    {
+      label: 'Status',
+      property: 'status',
+      width: '116px',
+    },
+    {
+      label: 'Activity',
+      visible: media.lg,
+      maxWidth: '15%',
+    },
+    {
+      label: 'Tags',
+      property: 'tags',
+      visible: media.md,
+      maxWidth: '15%',
+    },
+    {
+      label: 'Schedules',
+      property: 'schedules',
+      visible: media.md,
+      maxWidth: '15%',
+    },
+    {
+      label: 'Action',
+      width: '82px',
+    },
+  ])
+
+  const columnClasses = (column: TableColumn<Deployment>): ClassValue => [`deployment-list__${snakeCase(column.label)}-column`]
+  const rowClasses = (row: Deployment): ClassValue => {
+    return [
+      {
+        'deployment-list__row--subdued': row.paused,
+      },
+    ]
+  }
+
+  const selectedDeployments = ref<Deployment[]>([])
+
+  function refresh(): void {
+    subscriptions.refresh()
+  }
+
+  const emit = defineEmits<{
+    (event: 'delete'): void,
+  }>()
```

Comment:
nit: Nice to have all the "define" macros at the top of the script so you can see the component's api